### PR TITLE
Fix `DiGraph` edges not fading correctly on `FadeIn` and `FadeOut`

### LINKED ISSUE #3749: FadeOut on DiGraph does not work smoothly
## Description of bug / unexpected behavior
ManimCE v0.18.1

Using `self.play(Fadeout())` on a `DiGraph()` object does not correctly fade out the edges, only the nodes and arrow tips. Removing the individual subobjects works.

Link to the [Discord discussion](https://discord.com/channels/581738731934056449/1236723849937158145/1236723849937158145)

## Expected behavior
All elements of a `DiGraph()` should fade smoothly out.

## How to reproduce the issue
<details><summary>Code for reproducing the problem</summary>

```py
class IntroductionToGraphs(Scene):
    def construct(self):
        vertexs = [num for num in range(1,7)]
        arestes = [(1,2), (3,1), (1,5), (1,6), (5,1), (5,6), (3,4)]
        layout = "circular"

        graf_arestes1 = Graph(vertexs, arestes, labels = True, layout = layout)
        graf_arestes2 = DiGraph(vertexs, arestes, labels = True, layout = layout)
        graf_arestes3 = graf_arestes2.copy()

        graf_arestes1.scale_to_fit_width(14/3).to_edge(LEFT, buff=0)
        graf_arestes2.scale_to_fit_width(14/3)
        graf_arestes3.scale_to_fit_width(14/3).to_edge(RIGHT, buff=0)

        self.play(
            Create(graf_arestes1),
            Create(graf_arestes2),
            Create(graf_arestes3),
        )
        self.wait()
        self.play(
            FadeOut(graf_arestes1),
            FadeOut(graf_arestes2),
            FadeOut(*[obj for obj in graf_arestes3])
        )
```
</details>

## Additional media files
<details><summary>Images/GIFs</summary>

https://github.com/ManimCommunity/manim/assets/8582807/20f06405-6c81-453c-ac62-c6d32f16635c

</details>

## System specifications

<details><summary>System Details</summary>
irrelevant - the problem occurs on both Windows 10 and Manimator
</details>

### PR DESCRIPTION
**Pull request by Nikhil-42**
## Overview: What does this pull request change?
This pull request resolves issue #3749 in which edges do not FadeOut properly when using DiGraph. The default updater in the DiGraph class used the edge constructor `.become` to attach the edges to the vertices. I replaced the updater to function more similarly to the Graph updater which moves the endpoints of the Line. Also, noticed that `Line::set_points_by_ends` did not project onto the bounding box of `start` and `end` so I updated the behavior to match the constructor and the type hints to align with usage in the example scenes.

- Documented behavior of `Line::set_points_by_ends`
- Updated type hints in `Line`
- Altered edge updaters in `graph.py` to use `Line::set_points_by_ends` instead of `.become`

## Links to added or changed documentation pages

https://manimce--3786.org.readthedocs.build/en/3786/reference/manim.mobject.geometry.line.Line.html#manim.mobject.geometry.line.Line.set_points_by_ends

## Further Information and Comments

Fixes #3749 

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/geometry/line.py [hunk 1]
```diff
@@ -63,16 +63,32 @@ def generate_points(self) -> None:
...
+        Parameters
+        ----------
+        start : Point3D | Mobject
+            The start point or Mobject of the line.
+        end : Point3D | Mobject
+            The end point or Mobject of the line.
+        buff : float, optional
+            The empty space between start and end and the line, by default 0
+        path_arc : float, optional
+            The angle of a circle spanned by this arc, by default 0 which is a straight line.
```
Comment (chopan050) [inline manim/mobject/geometry/line.py:84]: When adding docstrings to a method, it is not necessary to repeat the parameter type hints inside the docstring: only add typings directly to the signature.

```suggestion
        start
            The start point or Mobject of the line.
        end
            The end point or Mobject of the line.
        buff
            The empty space between start and end and the line, by default 0
        path_arc
            The angle of a circle spanned by this arc, by default 0 which is a straight line.
```

**Review by chopan050 — CHANGES_REQUESTED** [review]
Thanks for your contribution! Looks good in general, just one small comment.

**Comment by Nikhil-42** [comment]
@chopan050 Thanks for the edit!

Side note: I'm interested in helping with the rewrite, do you know how I could get into that?

**Review by chopan050 — APPROVED** [review]
LGTM!

**Comment by chopan050** [comment]
> Side note: I'm interested in helping with the rewrite, do you know how I could get into that?

Any help is greatly appreciated! Simply join #dev-chat on the Manim Discord server, or the [manim/experimental] post on #development, and say that you're interested in helping. Somewhere in that post there's a list of the current tasks we have to do in experimental.

**Comment by JasonGrace2282** [comment]
Feel free to ping me on Discord (I'm still @JasonGrace2282 ) if you want me to walk you through where we are right now in the rewrite :)
I'm always glad to have more people interested

**Pull request merged**